haystack/testing/preview/document_store.py

```diff
@@ -17,8 +17,8 @@ def docstore(self) -> Store:
 
     @pytest.fixture
     def filterable_docs(self) -> List[Document]:
-        embedding_zero = np.zeros([768, 1]).astype(np.float32)
-        embedding_one = np.ones([768, 1]).astype(np.float32)
```

Comment:
not sure why this was defined as a matrix, vector databases usually expect vectors

Reply:
It is interesting that this didn't cause any error.
It is very easy introduce bugs like this when doing batching (either because of a batch size of 1 or wrong slicing of the batch). Do you think it would make sense if either the document store or the document constructor were to check the shape of the embedding so bugs like this are caught earlier?

Reply:
@MichelBartels yes, I think we should enforce not only the type but also the shape for the field `embedding`, we can make this part of the restructuring of `Document` which I want to schedule soon.